File: backend-spring/src/main/java/com/sdgp/backend/model/DataSet.java
package com.sdgp.backend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.util.Arrays;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Document("Datasets")
@TypeAlias("Dataset")
public class DataSet {
    @Id
    private int id;
    private String datasetName;
    private String publisher;
    private String year;
    private String releasedDate;
    private String category;
    private String description;
    private byte[] dataFile;

    @Override
    public String toString() {
        return "DataSet{" +
                "id='" + id + '\'' +
                ", datasetName='" + datasetName + '\'' +
                ", publisher='" + publisher + '\'' +
                ", year='" + year + '\'' +
                ", releasedDate='" + releasedDate + '\'' +
                ", category='" + category + '\'' +
                ", description='" + description + '\'' +
                ", dataFile=" + Arrays.toString(dataFile) +
                '}';
    }
}

File: frontend/src/app/models/dataset.ts
export class Dataset {
  id: number;
  datasetName: string;
  publisher: string;
  year: Date;
  releasedDate: Date;
  category: string;
  description: string;
}

export interface DataModel {
  District: string;
  number_sat: number;
  passed_all: number;
  passed_all_subject_perc: number;
  failed_all: number;
  failed_all_subject_perc: number;
}

export interface DataModel {
  District: string;
  number_sat: number;
  passed_all: number;
  passed_all_subject_perc: number;
  failed_all: number;
  failed_all_subject_perc: number;
}